### PR TITLE
Assignment Submission Report to Mentors

### LINKED ISSUE #467: Send assignment submission report to mentors
**Is your feature request related to a problem? Please describe.**

Upon the deadline of the submission of assignments, the mentors should get a submission report with at least the number of students who have submitted and those who missed the deadline. It will be added in a similar fashion as the [AssignmentDeadlineSubmissionJob](https://github.com/CircuitVerse/CircuitVerse/blob/master/app/jobs/assignment_deadline_submission_job.rb). Checkout [Jobs](https://guides.rubyonrails.org/active_job_basics.html) and [Mailers](https://guides.rubyonrails.org/v3.2/action_mailer_basics.html) in RubyOnRails to get started.

**Are you working on this? (Yes/No)**

No

### PR DESCRIPTION
**Pull request by NumberC**
Fixes #467

### CODE REVIEW

File: app/jobs/assignment_deadline_submission_job.rb [hunk 1]
```diff
@@ -3,13 +3,14 @@ class AssignmentDeadlineSubmissionJob < ApplicationJob
 
   def perform(assignment_id)
     assignment = Assignment.find_by(id:assignment_id)
+    @proj_s = 0
+    @proj_ns = 0
```
Comment (tachyons) [inline app/jobs/assignment_deadline_submission_job.rb]: Make variables for descriptive, like projects_sent. proj_sent is ambiguous and difficult to understand what is happening in the code

Comment (tachyons) [reply app/jobs/assignment_deadline_submission_job.rb]: Same for other newly introduced vars

Comment (NumberC) [reply app/jobs/assignment_deadline_submission_job.rb]: Shubhankar wanted me to resolve Hound issues. Many of them were about how long lines were. I had to drastically shorten variable names for that.

Comment (NumberC) [reply app/jobs/assignment_deadline_submission_job.rb]: Should I ignore Hound issues anyway and change the name?

Comment (tachyons) [reply app/jobs/assignment_deadline_submission_job.rb]: Hound is to make code make more readable, do not make code worse to silence hound. You can have longer names if that makes sense in that context, ignore hound errors in such contexts

**Comment by NumberC** [comment]
@tachyons Can you check it now?

**Comment by tachyons** [comment]
Other variables are still not descriptive enough

**Comment by NumberC** [comment]
@tachyons Sorry for tagging you so much, but I think now, it should be better.

File: app/jobs/assignment_deadline_submission_job.rb [hunk 2]
```diff
@@ -19,24 +20,23 @@ def perform(assignment_id)
...
               proj.assignment_id=nil
               proj.save!
               submission.save!
+            else
+              @projects_submitted += 1
             end
           end
           assignment.status = 'closed'
           assignment.save!
+          @asignment_mailer = AssignmentMailer.deadline_assignment_email(assignment, @projects_submitted, @projects_not_submitted)
```
[the remark this answers is not in the document]
Comment (tachyons) [reply app/jobs/assignment_deadline_submission_job.rb]: You can just call deliver later here directly, no need to assign it to a temp variable

File: spec/mailers/assignment_mailer_spec.rb [hunk 3]
```diff
@@ -26,4 +26,16 @@
       expect(mail.subject).to eq("Assignment Updated in #{@group.name}")
     end
   end
+
+  describe '#deadline_assignment_email' do
+    let(:mail) { AssignmentMailer.deadline_assignment_email(@assignment, @projects_submitted, @projects_not_submitted) }
+
+    it 'sends update assignment link' do
```
[the remark this answers is not in the document]
Comment (tachyons) [reply spec/mailers/assignment_mailer_spec.rb]: Handle this

**Comment by NumberC** [comment]
@tachyons Anything else?

File: app/mailers/assignment_mailer.rb [hunk 4]
```diff
@@ -10,4 +10,14 @@ def update_assignment_email(user,  assignment)
     @user = user
     mail(to: (@user.email), subject: "Assignment Updated in " +  Group.find_by(id:(@assignment.group_id)).name )
   end
+
+  def deadline_assignment_email(assignment, projects_submitted, projects_not_submitted)
+    @assignment = assignment
+    @projects_submitted = projects_submitted
+    @projects_not_submitted = projects_not_submitted
+    @user = User.find_by(id: Group.find_by(id: @assignment.group_id).mentor_id)
```
Comment (tachyons) [inline app/mailers/assignment_mailer.rb]: This should be something like

@user = @assignment.group.mentor

File: app/mailers/assignment_mailer.rb [hunk 5]
```diff
@@ -10,4 +10,14 @@ def update_assignment_email(user,  assignment)
...
+    @grp_name = Group.find_by(id: @assignment.group_id).name
```
Comment (tachyons) [inline app/mailers/assignment_mailer.rb]: @assignment.group.name

File: app/mailers/assignment_mailer.rb (hunk 4 above)
Comment (NumberC) [reply app/mailers/assignment_mailer.rb]: Thank you!

File: app/jobs/assignment_deadline_submission_job.rb [hunk 6]
```diff
@@ -19,24 +20,22 @@ def perform(assignment_id)
...
               proj.assignment_id=nil
               proj.save!
               submission.save!
+            else
+              @projects_submitted += 1
             end
           end
           assignment.status = 'closed'
           assignment.save!
+          AssignmentMailer.deadline_assignment_email(assignment, @projects_submitted, @projects_not_submitted).deliver_later
```
[the remark this answers is not in the document]
Comment (satu0king) [reply app/jobs/assignment_deadline_submission_job.rb]: This is an incorrect way to measure. `@assignment.projects` are all the projects that are getting submitted. The if condition is just a check. To check for those who are not submitted, you should see @group.group_members, that will be all members of the group and then see the difference in the count. Also add how many people are yet to join the group by checking pending invitations.

**Comment by satu0king** [comment]
@sal2701 can you check if this PR can be fixed?

**Comment by tachyons** [comment]
Closing due to inactivity

**Pull request closed without merging**